distributed-databases/pinot/addons.tf

```diff
@@ -0,0 +1,222 @@
+  eks_addons = {
+    aws-ebs-csi-driver = {
+      service_account_role_arn = module.ebs_csi_driver_irsa.iam_role_arn
+    }
+    coredns = {
+      preserve = true
+    }
+    vpc-cni = {
+      service_account_role_arn = module.vpc_cni_irsa.iam_role_arn
+      preserve                 = true
+    }
+    kube-proxy = {
+      preserve = true
+    }
+  }
+  #---------------------------------------
+  # Kubernetes Add-ons
+  #---------------------------------------
+
+  enable_kube_prometheus_stack = true
+  kube_prometheus_stack = {
+    namespace     = "monitoring"
+    name          = "prometheus"
+    chart_version = "48.1.1"
+    set_sensitive = [
+      {
+        name  = "grafana.adminPassword"
+        value = data.aws_secretsmanager_secret_version.admin_password_version.secret_string
+    }]
+
+    values = [
+      templatefile("${path.module}/helm/kube-prometheus-stack-values.yaml", {
+        storage_class_type = kubernetes_storage_class.ebs_csi_encrypted_gp3_storage_class.id,
+      })
+    ]
+  }
+
+  #---------------------------------------
+  # AWS for FluentBit - DaemonSet
+  #---------------------------------------
+  enable_aws_for_fluentbit = true
+  aws_for_fluentbit_cw_log_group = {
+    use_name_prefix   = false
+    name              = "/${local.name}/aws-fluentbit-logs" # Add-on creates this log group
+    retention_in_days = 30
+  }
+  aws_for_fluentbit = {
+    values = [templatefile("${path.module}/helm/aws-for-fluentbit-values.yaml", {
+      region               = local.region,
+      cloudwatch_log_group = "/${local.name}/aws-fluentbit-logs"
+      cluster_name         = module.eks.cluster_name
+    })]
+  }
+  #---------------------------------------
+  # AWS Load Balancer Controller
+  #---------------------------------------
+  enable_aws_load_balancer_controller = false
+
+
+  tags = local.tags
+}
+
+#---------------------------------------------------------------
+# Apache Pinot
+#---------------------------------------------------------------
+
+resource "random_string" "random_suffix" {
+  length  = 10
+  special = false
+  upper   = false
+}
+
+resource "random_password" "sensitive_key" {
+  length  = 16
+  special = false
+}
+
+module "eks_data_addons" {
+  source = "github.com/wahab-io/terraform-aws-eks-data-addons?ref=v1.2.6"
+
+  oidc_provider_arn = module.eks.oidc_provider_arn
+
+  #---------------------------------------------------------------
+  # Apache Pinot Add-on
+  #---------------------------------------------------------------
+  enable_pinot = true
+  pinot_helm_config = {
+    namespace = "pinot"
+    values    = [templatefile("${path.module}/helm/pinot-values.yaml", {})]
+    set = [
+      {
+        name  = "cluster.name"
+        value = local.cluster_name
+      },
+      {
+        name  = "controller.replicaCount"
+        value = 3
+      },
+      {
+        name  = "controller.persistence.storageClass"
+        value = "gp2"
```

Comment:
use gp3 since you are making this default

Reply:
Done